{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare 23andme SNPs imputed HDF5s from 1000G imputed VCFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bionc21\n",
      "Leipzig Cluster detected!\n",
      "/mnt/archgen/users/hringbauer/git/gedmatch_prep/python3\n",
      "CPU Count: 40\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import socket as socket\n",
    "import os as os\n",
    "import sys as sys\n",
    "import h5py\n",
    "import matplotlib.cm as cm\n",
    "import itertools as it\n",
    "import multiprocessing as mp\n",
    "socket_name = socket.gethostname()\n",
    "print(socket_name)\n",
    "\n",
    "if socket_name.startswith(\"bionc\"):\n",
    "    print(\"Leipzig Cluster detected!\")\n",
    "    path = \"/mnt/archgen/users/hringbauer/git/gedmatch_prep/python3/\"\n",
    "    #sys.path.append(\"/mnt/archgen/users/hringbauer/data/malta/\") \n",
    "    \n",
    "else: \n",
    "    raise RuntimeWarning(\"Not compatible machine. Check!!\")\n",
    "    \n",
    "os.chdir(path)  # Set the right Path (in line with Atom default)\n",
    "print(os.getcwd())\n",
    "print(f\"CPU Count: {mp.cpu_count()}\")\n",
    "\n",
    "from prep_23txt import prep_h5_autorun_23, generate_23_from_h5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1)  Prepare h5 (on 23andme v4&5 SNPs), and then 23andme txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running Chromosome 1...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch1.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch1.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 79239 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch1.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch1.h5\n",
      "\n",
      "Running Chromosome 2...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch2.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch2.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 82595 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch2.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch2.h5\n",
      "\n",
      "Running Chromosome 3...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch3.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch3.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 69195 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch3.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch3.h5\n",
      "\n",
      "Running Chromosome 4...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch4.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch4.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 63710 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch4.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch4.h5\n",
      "\n",
      "Running Chromosome 5...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch5.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch5.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 60823 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch5.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch5.h5\n",
      "\n",
      "Running Chromosome 6...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch6.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch6.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 70260 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch6.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch6.h5\n",
      "\n",
      "Running Chromosome 7...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch7.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch7.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 55823 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch7.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch7.h5\n",
      "\n",
      "Running Chromosome 8...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch8.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch8.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 53707 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch8.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch8.h5\n",
      "\n",
      "Running Chromosome 9...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch9.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch9.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 45137 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch9.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch9.h5\n",
      "\n",
      "Running Chromosome 10...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch10.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch10.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 51563 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch10.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch10.h5\n",
      "\n",
      "Running Chromosome 11...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch11.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch11.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 49149 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch11.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch11.h5\n",
      "\n",
      "Running Chromosome 12...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch12.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch12.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 48318 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch12.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch12.h5\n",
      "\n",
      "Running Chromosome 13...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch13.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch13.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 36737 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch13.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch13.h5\n",
      "\n",
      "Running Chromosome 14...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch14.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch14.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 32610 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch14.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch14.h5\n",
      "\n",
      "Running Chromosome 15...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch15.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch15.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 30507 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch15.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch15.h5\n",
      "\n",
      "Running Chromosome 16...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch16.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch16.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 32045 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch16.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch16.h5\n",
      "\n",
      "Running Chromosome 17...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch17.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch17.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 28401 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch17.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch17.h5\n",
      "\n",
      "Running Chromosome 18...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch18.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch18.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 29729 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch18.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch18.h5\n",
      "\n",
      "Running Chromosome 19...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch19.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch19.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 20504 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch19.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch19.h5\n",
      "\n",
      "Running Chromosome 20...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch20.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch20.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 24927 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch20.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch20.h5\n",
      "\n",
      "Running Chromosome 21...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch21.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch21.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 14473 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch21.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch21.h5\n",
      "\n",
      "Running Chromosome 22...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp45/CHC002/ch22.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme45/bcftools_ch22.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/CHC/CHC002/GTL_output/CHC002_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 14595 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch22.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/CHC002/ch22.h5\n",
      "\n",
      "Successfully saved 23andmefile to: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/txt23/CHC002.snps45.txt\n",
      "CPU times: user 33 s, sys: 3.6 s, total: 36.6 s\n",
      "Wall time: 15min 44s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "iid = \"CHC002\"\n",
    "prep_h5_autorun_23(iid=iid) # Generate h5\n",
    "\n",
    "### Convert to 23andme\n",
    "dft = generate_23_from_h5(chs =range(1,23), output=False, maxgp=0,\n",
    "        h5_path = f\"/mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp45/{iid}/\",\n",
    "        path_23tsv='/mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/snps45.tsv',\n",
    "        savepath =f\"/mnt/archgen/users/hringbauer/git/gedmatch_prep/output/txt23/{iid}.snps45.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Area 51"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "994884"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dft)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ch = dft[dft[\"chr\"]==str(3)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.8366863563495617"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(df_ch[\"gp\"]>=0.99)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
