{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare 23andme SNPs imputed HDF5s from 1000G imputed VCFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bionc21\n",
      "Leipzig Cluster detected!\n",
      "/mnt/archgen/users/hringbauer/brienzi\n",
      "CPU Count: 40\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import socket as socket\n",
    "import os as os\n",
    "import sys as sys\n",
    "import h5py\n",
    "import matplotlib.cm as cm\n",
    "import itertools as it\n",
    "import multiprocessing as mp\n",
    "\n",
    "socket_name = socket.gethostname()\n",
    "print(socket_name)\n",
    "\n",
    "    \n",
    "if socket_name.startswith(\"bionc\"):\n",
    "    print(\"Leipzig Cluster detected!\")\n",
    "    path = \"/mnt/archgen/users/hringbauer/brienzi/\"\n",
    "    #sys.path.append(\"/mnt/archgen/users/hringbauer/data/malta/\") \n",
    "    \n",
    "else: \n",
    "    raise RuntimeWarning(\"Not compatible machine. Check!!\")\n",
    "    \n",
    "os.chdir(path)  # Set the right Path (in line with Atom default)\n",
    "print(os.getcwd())\n",
    "print(f\"CPU Count: {mp.cpu_count()}\")\n",
    "\n",
    "from ancIBD.IO.prepare_h5 import vcf_to_1240K_hdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Todo: Write code to automatize hdf5 production with 23andme SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_path_imputed1000g_autorun(iid =\"\"):\n",
    "    \"\"\"Get path of imputed VCF (all 1000G variants) of iid\n",
    "    Return this path\"\"\"\n",
    "    path_imp_vcf = f\"/mnt/archgen/Autorun_eager/eager_outputs/TF/{iid[:3]}/{iid}/GTL_output/{iid}_imputed.vcf.gz\" \n",
    "    assert(os.path.exists(path_imp_vcf))\n",
    "    return path_imp_vcf\n",
    "\n",
    "def prep_h5_autorun_23(iid=\"\"):\n",
    "    \"\"\"Prepare HDF5 File for Individual IID in autorun.\n",
    "    Hard Coded MPI EVA Folder Structure.\n",
    "    Todo: Create VCF and HDF output path if not existing...\"\"\"\n",
    "    path_vcf_imp = get_path_imputed1000g_autorun(iid =iid)\n",
    "    \n",
    "    for ch in range(1,23):\n",
    "        print(f\"Running Chromosome {ch}...\")\n",
    "        vcf_to_1240K_hdf(in_vcf_path = path_vcf_imp,\n",
    "                 path_vcf = f\"/mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/{iid}/ch{ch}.vcf.gz\",\n",
    "                 path_h5 = f\"/mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/{iid}/ch{ch}.h5\",\n",
    "                 marker_path = f\"/mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch{ch}.csv\",\n",
    "                 map_path = \"\", \n",
    "                 af_path = \"\",\n",
    "                 col_sample_af = \"AF_SAMPLE\",\n",
    "                 ch=ch, \n",
    "                 chunk_length=10000)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare H5 of Bishop Wichmann"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running Chromosome 1...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch1.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch1.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 41731 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch1.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch1.h5\n",
      "\n",
      "Running Chromosome 2...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch2.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch2.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 42728 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch2.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch2.h5\n",
      "\n",
      "Running Chromosome 3...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch3.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch3.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 35511 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch3.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch3.h5\n",
      "\n",
      "Running Chromosome 4...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch4.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch4.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 32175 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch4.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch4.h5\n",
      "\n",
      "Running Chromosome 5...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch5.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch5.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 31880 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch5.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch5.h5\n",
      "\n",
      "Running Chromosome 6...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch6.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch6.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 36919 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch6.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch6.h5\n",
      "\n",
      "Running Chromosome 7...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch7.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch7.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 29156 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch7.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch7.h5\n",
      "\n",
      "Running Chromosome 8...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch8.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch8.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 28828 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch8.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch8.h5\n",
      "\n",
      "Running Chromosome 9...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch9.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch9.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 25099 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch9.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch9.h5\n",
      "\n",
      "Running Chromosome 10...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch10.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch10.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 27662 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch10.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch10.h5\n",
      "\n",
      "Running Chromosome 11...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch11.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch11.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 25715 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch11.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch11.h5\n",
      "\n",
      "Running Chromosome 12...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch12.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch12.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 25830 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch12.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch12.h5\n",
      "\n",
      "Running Chromosome 13...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch13.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch13.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 19808 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch13.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch13.h5\n",
      "\n",
      "Running Chromosome 14...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch14.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch14.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 17291 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch14.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch14.h5\n",
      "\n",
      "Running Chromosome 15...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch15.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch15.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 16249 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch15.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch15.h5\n",
      "\n",
      "Running Chromosome 16...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch16.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch16.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 16743 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch16.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch16.h5\n",
      "\n",
      "Running Chromosome 17...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch17.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch17.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 14802 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch17.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch17.h5\n",
      "\n",
      "Running Chromosome 18...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch18.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch18.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 15812 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch18.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch18.h5\n",
      "\n",
      "Running Chromosome 19...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch19.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch19.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 10130 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch19.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch19.h5\n",
      "\n",
      "Running Chromosome 20...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch20.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch20.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 13644 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch20.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch20.h5\n",
      "\n",
      "Running Chromosome 21...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch21.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch21.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 7896 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch21.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch21.h5\n",
      "\n",
      "Running Chromosome 22...\n",
      "Print downsampling to 1240K...\n",
      "Running bash command: \n",
      "bcftools view -Oz -o /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/vcfs.23imp/WSQ001/ch22.vcf.gz -T /mnt/archgen/users/hringbauer/git/gedmatch_prep/data/23andme/bcftools_ch22.csv -M2 -v snps /mnt/archgen/Autorun_eager/eager_outputs/TF/WSQ/WSQ001/GTL_output/WSQ001_imputed.vcf.gz\n",
      "Finished BCF tools filtering to target markers.\n",
      "Converting to HDF5...\n",
      "Finished conversion to hdf5!\n",
      "Merging in LD Map..\n",
      "Calculating in sample allele frequencies and saving at hdf5 column AF_SAMPLE\n",
      "Adding Allele Frequencies to HDF5...\n",
      "Loaded 8021 variants.\n",
      "Finshed merged in allele frequencies into /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch22.h5\n",
      "Transformation complete! Find new hdf5 file at: /mnt/archgen/users/hringbauer/git/gedmatch_prep/output/h5.23imp/WSQ001/ch22.h5\n",
      "\n",
      "CPU times: user 4.64 s, sys: 192 ms, total: 4.84 s\n",
      "Wall time: 15min 36s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "prep_h5_autorun_23(iid=\"WSQ001\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
